src/osqp/tests/derivative_test.py

```diff
@@ -51,7 +56,7 @@ def get_grads(self, P, q, A, l, u, true_x, true_yl=None, true_yu=None, mode='qdl
         # Get gradients by solving with osqp
         m = osqp.OSQP()
         m.setup(P, q, A, l, u, eps_abs=eps_abs, eps_rel=eps_rel,
-                max_iter=max_iter, verbose=True)
+                max_iter=max_iter, verbose=True, scaling=0)
```

Comment:
I'm getting different results on the C side for `P` values unless I set `scaling=0`. If arbitrary scaling needs to be supported, perhaps we should make an issue on this (and see where in the C code we need to tweak things).

Reply:
It makes sense that scaling has a special effect on the C side since data is rescaled there. However, the C side **should unscale the results before returning them**. Otherwise the differentiation will not work.

For the backward (adjoint) derivative, the results have the shape of the data. I suspect you just need to unscale them using the [unscale_data](https://github.com/osqp/osqp/blob/0b34f2ef5c5eec314e7945762e1c8167e937afbd/src/scaling.c#L160) function. (and use (dP, dq, dA, dl, du) instead of (P, q, A, l, u).

For the forward derivative, I believe you should use [unscale_solution](https://github.com/osqp/osqp/blob/0b34f2ef5c5eec314e7945762e1c8167e937afbd/src/scaling.c#L177) over (dx, dy) instead of (x, y).

Reply:
Thanks @bstellato. This is very helpful. Let me save these comments in a new issue and merge/close this PR so it's not sitting around